packages/core/content-manager/server/src/history/controllers/history-version.ts

```diff
@@ -60,21 +61,35 @@ const createHistoryVersionController = ({ strapi }: { strapi: Core.Strapi }) =>
         return ctx.forbidden();
       }
 
-      const params: HistoryVersions.GetHistoryVersions.Request['query'] =
+      const query: HistoryVersions.GetHistoryVersions.Request['query'] =
         await permissionChecker.sanitizeQuery(ctx.query);
 
       const { results, pagination } = await getService(strapi, 'history').findVersionsPage({
-        ...params,
-        ...getValidPagination({ page: params.page, pageSize: params.pageSize }),
+        query: {
+          ...query,
+          ...getValidPagination({ page: query.page, pageSize: query.pageSize }),
+        },
+        state: { userAbility: ctx.state.userAbility },
       });
 
+      const sanitizedResults = await async.map(
+        results,
+        async (version: HistoryVersions.HistoryVersionDataResponse & { locale: string }) => {
+          return {
+            ...version,
+            data: await permissionChecker.sanitizeOutput({
+              ...version.data,
+              locale: version.locale.code,
```

Comment:
I ahve checked the sanitization code and I don't believe there is anything specific about needing the locale code here 🤔 

In my opinion all fields should be available in version.data already, and its format should be enough for the sanitizer to take care of any field that should be filtered out